docs/references/ic-interface-spec.md

```diff
@@ -1390,6 +1394,18 @@ While an implementation will likely try to keep the interval between the value o
 
 :::
 
+#### On Low Wasm Memory {#on-low-wasm-memory}
+
+A canister can export a function with the name `canister_on_low_wasm_memory`, which is scheduled whenever the canister's remaining wasm memory size in bytes falls from at least a threshold `t` to strictly less than `t`.
+The threshold `t` can be defined in the [canister's settings](#ic-canister_status) and by default it is set to 0.
+
+:::note
+
+While the above function is scheduled immediately once the condition above is triggered, it may not necessarily be executed immediately if the canister does not have enough cycles.
+If the canister gets frozen immediately after the function is scheduled for execution, the function will run once the canister's unfrozen _if_ the canister's wasm memory size in bytes remains strictly less than the threshold `t`.
```

Comment:
Another case in which canister may not be executed immediately is if the used_wasm_memory becomes greater than wasm_memory_limit, in which case canister will execute the hook when wasm_memory_limit is increased here is [PR](https://github.com/dfinity/ic/pull/3631) that implemented that logic.

Reply:
Thank you for the heads up! I've left a question on the PR. Based on your reply there and a potential discussion in the Interface Spec meeting, I'll come back to you.

Reply:
Updated the spec: https://github.com/dfinity/portal/pull/3761/commits/6d37ea3f3ad650c44c6a7447bb040fb31124a860